app/src/main/java/com/example/tasky/feature_agenda/presentation/event_detail_screen/EventDetailViewModel.kt

```diff
@@ -0,0 +1,194 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.util.UUID
+import javax.inject.Inject
+
+@HiltViewModel
+class EventDetailViewModel @Inject constructor(
+    private val useCases: AgendaUseCases,
+    private val repositories: AgendaRepositories,
+    private val savedStateHandle: SavedStateHandle,
+    private val userPreferences: UserPreferences
+): ViewModel() {
+
+    private val _state = MutableStateFlow(EventDetailState())
+    val state = _state.asStateFlow()
+
+    private val photoList = mutableStateListOf<Photo>()
+
+    init {
+        val eventId = savedStateHandle.get<String>(ArgumentTypeEnum.ITEM_ID.name)
+        val editMode = savedStateHandle.get<String>(ArgumentTypeEnum.EDIT_MODE.name)
+
+        eventId?.let {
+            val editable = editMode != null
+            getSelectedEvent(it, editable)
+        }
+    }
+
+    fun onEvent(event: EventDetailOnClick) {
+        when(event) {
+            EventDetailOnClick.SaveEvent -> {
+                //saveEvent()
+            }
+            is EventDetailOnClick.SelectFilterOption -> {
+                _state.update {
+                    it.copy(selectedChip = event.selectedChip)
+                }
+            }
+            EventDetailOnClick.ToggleAddingAttendeeDialog -> {
+                _state.update {
+                    it.copy(addingAttendees = !it.addingAttendees)
+                }
+            }
+            EventDetailOnClick.ToggleAddingPhotos -> {
+                _state.update {
+                    it.copy(addingPhotos = !it.addingPhotos)
+                }
+            }
+            EventDetailOnClick.ToggleEditMode -> {
+                _state.update {
+                    it.copy(editMode = !it.editMode)
+                }
+            }
+            EventDetailOnClick.ToggleReminderMenu -> {
+                _state.update {
+                    it.copy(isReminderMenuVisible = !it.isReminderMenuVisible)
+                }
+            }
+
+            is EventDetailOnClick.DescriptionChanged -> {
+                _state.update {
+                    it.copy(eventDescription = event.description)
+                }
+            }
+            is EventDetailOnClick.TitleChanged -> {
+                _state.update {
+                    it.copy(eventTitle = event.title)
+                }
+            }
+
+            is EventDetailOnClick.AddPhoto -> {
+                if(event.photoUri == null) {
+                    _state.update {
+                        it.copy(addingPhotos = !it.addingPhotos)
+                    }
+                } else {
+                    photoList.add(
+                        Photo(
+                            key = UUID.randomUUID().toString(),
+                            url = event.photoUri.toString()
```

Comment:
A Uri is not really a URL(URL = web address, URI = pointer to a resource). A content Uri on Android therefore is no URL. I've handled it like this:
```
@Parcelize
sealed interface EventPhoto: Parcelable {

    @Parcelize
    data class Remote(
        val key: String,
        val photoUrl: String,
    ) : EventPhoto, Parcelable

    @Parcelize
    class Local(
        val key: String,
        val uri: Uri,
    ) : EventPhoto, Parcelable
}
```